File: src/main/java/responsemodels/QueryResponseMessage.java
package responsemodels;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import javax.annotation.Nonnull;

public class QueryResponseMessage {

  @SerializedName("data")
  @Expose
  private Data data;

  public Data getData() {
    return data;
  }

  public void setData(Data data) {
    this.data = data;
  }

  public static class Data {
    public Data(String responseMsg) {
      this.responseMsg = responseMsg;
    }

    @SerializedName("responseMsg")
    @Expose
    private String responseMsg;

    public String getResponseMsg() {
      return responseMsg;
    }

    public void setResponseMsg(String responseMsg) {
      this.responseMsg = responseMsg;
    }
  }

  public static class Builder implements QueryResponseBuilder {
    private final QueryResponseMessage queryResponseMessage;

    public Builder() {
      this.queryResponseMessage = new QueryResponseMessage();
    }

    @Nonnull
    @Override
    public QueryResponseBuilder setResponseMessage(@Nonnull String queryResponseMsg) {
      this.queryResponseMessage.setData(new Data(queryResponseMsg));
      return this;
    }

    @Nonnull
    @Override
    public QueryResponseMessage build() {
      return this.queryResponseMessage;
    }
  }
}

File: src/main/java/responsemodels/QueryResponseMessage.kt
package responsemodels

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class QueryResponseMessage(
    @SerializedName("data")
    @Expose
    var data: Data = Data()
) {
    data class Data(@field:Expose @field:SerializedName("responseMsg") var responseMsg: String = "")
    class QueryResponseMessageBuilder : QueryResponseBuilder {
        private val queryResponseMessage: QueryResponseMessage = QueryResponseMessage()

        override fun setResponseMessage(queryResponseMsg: String): QueryResponseBuilder {
            queryResponseMessage.data.responseMsg = queryResponseMsg
            return this
        }

        override fun build(): QueryResponseMessage {
            return queryResponseMessage
        }
    }
}